### ISSUE TITLE
Bun fails with "ReferenceError" for enclosed function variable

### ISSUE THREAD

**Issue by ivan-kleshnin** [oven-sh/bun#13208]
### What version of Bun is running?

1.1.22

### What platform is your computer?

Darwin 23.5.0 arm64 arm

### What steps can reproduce the bug?

I'm trying to replace Node.js with Bun. Seing a lot of false positive reports of `ReferenceError: can't find variable: ...`.
Here's one example, reduced to a sandbox size:

```ts
import {once} from "node:events"

async function getResource() {
  return {
    [Symbol.dispose]: () => {
      console.log("Dispose")
    }
  }
}

using resource = await getResource()

let stopped = false
async function run() {
  while (!stopped) { // Bun crashes with "ReferenceError: Can't find variable: stopped"
    console.log(new Date())
    await new Promise(resolve => setTimeout(resolve, 1000))
  }
}

await Promise.race([
  run(),
  once(process, "SIGINT"),
])

stopped = true

console.log("End")
```

The above code works perfectly fine in Node.js (e.g. using `tsx` library to run TypeScript OR in raw Node.js if type syntax is removed). 

To generalize:
- It usually happens with `async` functions that for no clear reason lose acccess to variables in a closure.
- It happens with local variables, I've noticed no similar issues with imported values / types so far

**Comment by Jarred-Sumner** [comment]
This is a transpiler bug with `using`

**Comment by ivan-kleshnin** [comment]
Yes, it might be that. I can't find a related thread on GitHub because "using" is a super frequent word, I'd have to inspect hundreds of them 😅  Could you please share a link to the issue you have in mind?

**Comment by nektro** [comment]
fixed in bun@1.1.31
